### PR TITLE
Obfuscate the GlobalOpenTelemetry instance.

### LINKED ISSUE #2788: Obfuscate GlobalOpenTelemetry.get()
I'm nervous about users casting the return value of `GlobalOpenTelemetry.get()` to `OpenTelemetrySdk`, e.g.

```
((OpenTelemetrySdk) GlobalOpenTelemetry.get()).getSdkTracerProvider()...
```

It seems like a reasonable way to get the global SDK now that we no longer have `OpenTelemetrySdk.get()`, so I suspect users will find their way to doing it.

The reason I'm nervous about this, is that the javaagent returns a bridged OpenTelemetry instance from GlobalOpenTelemetry.get(), which will then cause the user's app to throw ClassCastException, and a (the?) top priority of the java agent is to not break a user's application.

This is similar to the reason the return value from `GlobalOpenTelemetry.get().getTracerProvider()` is obfuscated, in order to prevent similar global access to the SDK:

```
((SdkTracerProvider) GlobalOpenTelemetry.get().getTracerProvider())...
```

### PR DESCRIPTION
**Pull request by jkwatson**
This is to prevent people from casting to the SDK implementation.
Resolves #2788

### CODE REVIEW

**Review by trask — APPROVED** [review]
Thanks 🙏

**Review by carlosalberto — APPROVED** [review]

File: api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java [hunk 1]
```diff
@@ -166,4 +167,29 @@ private static OpenTelemetry maybeAutoConfigure() {
...
+      this.delegate = delegate;
+    }
+
+    @Override
+    public TracerProvider getTracerProvider() {
+      return delegate.getTracerProvider();
+    }
+
+    @Override
+    public ContextPropagators getPropagators() {
```
Comment (anuraaga) [inline api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java:191]: Can you add this to the test too?

File: sdk/all/src/test/java/io/opentelemetry/sdk/OpenTelemetrySdkTest.java [hunk 2]
```diff
@@ -44,10 +44,10 @@ void tearDown() {
...
     OpenTelemetrySdk sdk = OpenTelemetrySdk.builder().buildAndRegisterGlobal();
-    assertThat(sdk).isSameAs(GlobalOpenTelemetry.get());
-    assertThat(GlobalOpenTelemetry.get()).isSameAs(sdk);
-    assertThat(((OpenTelemetrySdk) GlobalOpenTelemetry.get()).getSdkTracerProvider().get(""))
-        .isSameAs(GlobalOpenTelemetry.getTracerProvider().get(""));
+    assertThat(GlobalOpenTelemetry.get()).extracting("delegate").isSameAs(sdk);
+    assertThat(sdk.getTracerProvider().get(""))
+        .isSameAs(GlobalOpenTelemetry.getTracerProvider().get(""))
+        .isSameAs(GlobalOpenTelemetry.get().getTracer(""));
   }
```
Comment (anuraaga) [inline sdk/all/src/test/java/io/opentelemetry/sdk/OpenTelemetrySdkTest.java:51]: Can you add a test that attempts to cast the SDK?

File: api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java (hunk 1 above)
Comment (jkwatson) [reply api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java:191]: done

File: sdk/all/src/test/java/io/opentelemetry/sdk/OpenTelemetrySdkTest.java (hunk 2 above)
Comment (jkwatson) [reply sdk/all/src/test/java/io/opentelemetry/sdk/OpenTelemetrySdkTest.java:70]: done

**Review by anuraaga — APPROVED** [review]

**Pull request merged**